### PR TITLE
minizip-ng: add version 4.0.7

### PR DESCRIPTION
**Pull request by toge**
### Summary
Changes to recipe:  **minizip-ng/4.0.7**

#### Motivation
There is a new feature in 4.0.7.

#### Details
https://github.com/zlib-ng/minizip-ng/releases/tag/4.0.7
https://github.com/zlib-ng/minizip-ng/compare/4.0.6...4.0.7

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

File: recipes/minizip-ng/all/conanfile.py [hunk 1]
```diff
@@ -162,7 +162,7 @@ def package_info(self):
         self.cpp_info.set_property("pkg_config_name", "minizip")
 
         # TODO: back to global scope in conan v2 once cmake_find_package_* generators removed
-        prefix = "lib" if is_msvc(self) or self._is_clang_cl else ""
+        prefix = "lib" if Version(self.version) < "4.0.7" and (is_msvc(self) or self._is_clang_cl) else ""
```
Comment (proceduralnoisy) [inline recipes/minizip-ng/all/conanfile.py:165]: For anyone else curious, the corresponding minizip-ng change is [here](https://github.com/zlib-ng/minizip-ng/commit/b887f8104bceaf0d6cd0bb3f24e6f17eb68cb73d).

**Review by proceduralnoisy — APPROVED** [review]
Thanks for tackling this one! I happened to need it today, so was glad to see your PR.

The changes look good to me. I have locally tested this recipe (MSVC 2022 with `options={"mz_compatibility": True}`). I can confirm that new functionality introduced in 4.0.7 works as expected.

**Review by ErniGH — APPROVED** [review]
LGTM Thanks for your effort 😄

File: recipes/minizip-ng/all/conanfile.py (hunk 1 above)
Comment (ErniGH) [reply recipes/minizip-ng/all/conanfile.py:165]: Thanks for your help 😄

**Pull request merged**